kola/tests/misc/raid.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package misc
+
+import (
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform/conf"
+)
+
+func init() {
+	register.Register(&register.Test{
+		Run:         RootOnRaid,
+		ClusterSize: 1,
+		Name:        "coreos.misc.root-on-raid",
+		// This will only work on qemu, since it's overwriting
+		// /usr/share/oem/grub.cfg
+		Platforms: []string{"qemu"},
+		UserData: conf.ContainerLinuxConfig(`storage:
+  raid:
+    - name: "ROOT"
+      level: "raid1"
+      devices:
+        - "/dev/disk/by-partlabel/ROOT"
+        - "/dev/disk/by-partlabel/USR-B"
+  filesystems:
+    - name: "ROOT"
+      mount:
+        device: "/dev/md127"
```

Comment:
I'm not seeing anything in `/dev/disk` at the very least, so I'm not sure what else to do here other than maybe suggest Ignition gains the ability to parse `/etc/mdstat`